packages/core/src/options.ts

```diff
@@ -44,18 +44,21 @@ export interface Emoji {
 	embedEmoji?: boolean;
 }
 
-const serverSideSafeWindow = window ?? globalThis;
+// We need to test for typeof window !== 'undefined' because of how most SSR frameworks work.
+const serverSideSafeWindow = typeof window !== 'undefined' && window !== undefined && window !== null ? window : globalThis;
```

Comment:
```suggestion
const serverSideSafeWindow = typeof window === 'undefined' || !window ? globalThis : window;
```

Slight adjustment to avoid [no-negated-condition](https://eslint.org/docs/latest/rules/no-negated-condition) pattern